src/pages/saksbehandling/regulering/ManuellRegulering.tsx

```diff
@@ -0,0 +1,283 @@
+                }
+            ),
+    });
+
+    useEffect(() => {
+        hentGjeldendeGrunnlagsdataOgVilkårsvurderinger(
+            {
+                sakId: props.sak.id,
+                fraOgMed: regulering.periode.fraOgMed,
+            },
+            (data) =>
+                formik.setValues({
+                    uføre:
+                        data.grunnlagsdataOgVilkårsvurderinger.uføre?.vurderinger
+                            .map((v) => v?.grunnlag)
+                            .filter(filtrerRegulerbarIEU) ?? [],
+                    fradrag: fjernFradragSomIkkeErVelgbareEkskludertNavYtelserTilLivsopphold(
+                        data.grunnlagsdataOgVilkårsvurderinger.fradrag
+                    ).map(fradragTilFradragFormData),
+                })
+        );
+    }, []);
+
+    return pipe(
+        gjeldendeGrunnlagsdataOgVilkårsvurderinger,
+        RemoteData.fold(
+            () => <Loader />,
+            () => <Loader />,
+            () => null,
+            (gjeldendeVedtaksdata) => {
+                const { uføre, fradrag } = gjeldendeVedtaksdata.grunnlagsdataOgVilkårsvurderinger;
+                const uføregrunnlag = uføre?.vurderinger.map((v) => v?.grunnlag).filter(filtrerRegulerbarIEU) ?? [];
+                const harRegulerbarIEU = uføregrunnlag.some((v) => v.forventetInntekt > 0);
+                const harRegulerbarFradrag = fradrag.some((f) => måReguleresManuelt(f.type));
+
+                const hentProblemer = (årsaker: ÅrsakForManuell[]) =>
+                    årsaker.filter(
+                        (årsak) =>
+                            årsak !== ÅrsakForManuell.ForventetInntektErStørreEnn0 &&
+                            årsak !== ÅrsakForManuell.FradragMåHåndteresManuelt
+                    );
+
+                const hentTekstForManuellÅrsak = (årsak: ÅrsakForManuell): string => {
+                    switch (årsak) {
+                        case ÅrsakForManuell.FradragMåHåndteresManuelt:
+                        case ÅrsakForManuell.ForventetInntektErStørreEnn0:
+                            return '';
+
+                        case ÅrsakForManuell.YtelseErMidlertidigStanset:
+                            return formatMessage('manuell.årsak.stans');
+                        case ÅrsakForManuell.DelvisOpphør:
+                        case ÅrsakForManuell.VedtakstidslinjeErIkkeSammenhengende:
+                            return formatMessage('manuell.årsak.hull');
+                        case ÅrsakForManuell.PågåendeAvkortingEllerBehovForFremtidigAvkorting:
+                            return formatMessage('manuell.årsak.avkorting');
+                        case ÅrsakForManuell.AvventerKravgrunnlag:
+                            return formatMessage('manuell.årsak.avventarKravgrunnlag');
+                        case ÅrsakForManuell.UtbetalingFeilet:
+                            return formatMessage('manuell.årsak.utbetalingFeilet');
+                    }
+                };
```

Comment:
Disse to funksjonene bør ikke ligge i view-logikken, men hvert fall trekkes ut av `pipe`. Kan kanskje ligge utenfor komponentet? (men fortsatt i samme fil)

Reply:
Trekker det ut!